src/foundry/foundry.js/canvas.d.ts

```diff
@@ -17,401 +20,498 @@
+     */
+    protected _onDragLeftDrop(
+      event: PIXI.InteractionEvent
+    ): ReturnType<PlaceablesLayer<any>['selectObjects']> | ReturnType<TokenLayer['targetObjects']> | void;
+
+    /**
+     * Handle the cancellation of a left-mouse drag workflow
+     * @see {@link MouseInteractionManager#_handleDragCancel}
+     */
+    protected _onDragLeftCancel(event: PointerEvent): PIXI.Graphics | void;
+
+    /**
+     * Handle right mouse-click events occurring on the Canvas stage or it's active layer
+     * @see {@link MouseInteractionManager#_handleClickRight}
+     */
+    protected _onClickRight(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Handle right-mouse drag events occuring on the Canvas stage or an active Layer
+     * @see {@link MouseInteractionManager#_handleDragMove}
+     */
+    protected _onDragRightMove(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Handle the conclusion of a right-mouse drag workflow the Canvas stage.
+     * @see {@link MouseInteractionManager#_handleDragDrop}
+     * @param event - (unused)
+     */
+    protected _onDragRightDrop(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Determine selection coordinate rectangle during a mouse-drag workflow
+     */
+    protected _onDragSelect(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Pan the canvas view when the cursor position gets close to the edge of the frame
+     * @param event - The originating mouse movement event
+     */
+    protected _onDragCanvasPan(event: MouseEvent): ReturnType<this['animatePan']> | void;
+
+    /**
+     * Handle window resizing with the dimensions of the window viewport change
+     * @param event - The Window resize event
+     *                (default: `null`)
+     */
+    protected _onResize(event?: UIEvent | null): false | void;
+
+    /**
+     * Handle mousewheel events which adjust the scale of the canvas
+     * @param event - The mousewheel event that zooms the canvas
+     */
+    protected _onMouseWheel(event: WheelEvent): void;
+
+    /**
+     * Event handler for the drop portion of a drag-and-drop event.
+     */
+    protected _onDrop(event: DragEvent): unknown;
+
+    /**
+     * Add a pending canvas operation that should fire once the socket handling workflow concludes.
+     * This registers operations by a unique string name into a queue - avoiding repeating the same work multiple times.
+     * This is especially helpful for multi-object updates to avoid costly and redundant refresh operations.
+     * @param name  - A unique name for the pending operation, conventionally Class.method
+     * @param fn    - The unbound function to execute later
+     * @param scope - The scope to which the method should be bound when called
+     * @param args  - Arbitrary arguments to pass to the method when called
+     */
+    addPendingOperation<S = any, A = any[]>(name: string, fn: (this: S, args: A) => void, scope: S, args: A): void;
```

Comment:
no need to provide defaults for the generic parameters, they should be inferred from the runtime arguments.